### ISSUE TITLE
Value RetentionInDays only accepts certain numbers

### ISSUE THREAD

**Issue by jacobdonenfeld** [serverless/serverless#11495]
### Is there an existing issue for this?

- [X] I have searched existing issues, it hasn't been reported yet

### Use case description

When defining provider: RetentionInDays, only values 1,3,5,7,14,30,60,90,120,150,180,365,400,545,731,1827,3653 are accepted, not any integer. My intuition makes me think this accepts any integer. 

### Proposed solution (optional)

I'd like to update the example where this is defined and provide a comment stating acceptable values.

**Comment by pgrzesik** [comment]
Hey @jacobdonenfeld - that's not true, based on CloudFormation docs, it's only possible to specify a predefined values, not any integer: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-loggroup.html

**Comment by jacobdonenfeld** [comment]
@pgrzesik Yeah, I'd like to add a comment stating the valid values, not support any integer. Worth adding in a comment in the example, or a link to valid log retention values (to keep the documentation in one place? 
https://github.com/serverless/serverless/pull/11496

**Comment by pgrzesik** [comment]
I see @jacobdonenfeld, that makes sense 👍 I think it's worth adding a reference to official CloudFormation docs in the PR as well 👍
